todo-app/src/app/todo-list/todo-list.component.ts

```diff
@@ -0,0 +1,178 @@
+import {Component, OnInit, ViewEncapsulation} from '@angular/core';
+import { ITodo } from "../Models/todo.model";
+import { TodoDataService } from "../Services/todo-data.service";
+import { Todo } from "../Models/todo";
+import { FormControl, FormGroup, Validators } from "@angular/forms";
+import { CdkDragDrop, moveItemInArray } from "@angular/cdk/drag-drop";
+
+import {MomentDateAdapter, MAT_MOMENT_DATE_ADAPTER_OPTIONS} from '@angular/material-moment-adapter';
+import {DateAdapter, MAT_DATE_FORMATS, MAT_DATE_LOCALE} from '@angular/material/core';
+import * as _moment from 'moment';
+import { default as _rollupMoment } from 'moment';
+
+import {trigger, keyframes, animate, transition} from "@angular/animations";
+import * as keyfr from '../keyframes/keyframes';
+const moment = _rollupMoment || _moment;
+
+
+
+export const MY_FORMATS = {
+  parse: {
+    dateInput: 'LL',
+  },
+  display: {
+    dateInput: 'LL',
+    monthYearLabel: 'MMM YYYY',
+    dateA11yLabel: 'LL',
+    monthYearA11yLabel: 'MMMM YYYY',
+  },
+};
+
+@Component({
+  selector: 'app-todo-list',
+  templateUrl: './todo-list.component.html',
+  styleUrls: ['./todo-list.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+  providers: [ TodoDataService,
+    {
+      provide: DateAdapter,
+      useClass: MomentDateAdapter,
+      deps: [MAT_DATE_LOCALE, MAT_MOMENT_DATE_ADAPTER_OPTIONS]
+    },
+
+    {provide: MAT_DATE_FORMATS, useValue: MY_FORMATS},
+  ],
+  animations: [
+    trigger( 'noteAnimation', [
+      transition('* => swipeLeft', animate(700, keyframes(keyfr.swipeLeft))),
+      transition('* => swipeRight', animate(700, keyframes(keyfr.swipeRight))),
+    ])
+  ]
+})
+
+export class TodoListComponent implements OnInit {
+
+  todos: ITodo[];
+  todo: ITodo = new Todo(  "", "", false, '');
+  form: FormGroup
+  list: FormGroup
+  searchInput: string
+  step = 0
+  animationState: string
+
+  constructor(private todoDataService: TodoDataService) { }
+
+  ngOnInit(): void {
+    this.form = new FormGroup({
```

Comment:
Move creation FormGroup to the new method.

Reply:
done